smile_id_core/web_api.py

```diff
@@ -106,13 +105,36 @@ def verify_document(
         if last_name:
             payload["last_name"] = last_name
 
-        response = self._make_request("POST", self.Urls.VERIFY_DOCUMENT, payload)
+        return payload
+
+    def verify_document(self, **data):
+        """
+            Performs a document verification request with an immediate response.
+             **data consists of parameters for _prepare_document_verification_payload() method above to create payload
+             for request.
+            :return: dict
+        """
+        payload = self._prepare_document_verification_payload(**data)
+        response = self._make_request("POST", self.Urls.DOCUMENT_VERIFICATION,
+                                      payload)
 
         assert isinstance(response, dict)
         if response.get("ResultCode") != StatusCodes.SUCCESS:
             raise VerificationFailed(response)
         return response
 
+    def verify_document_async(self, **data):
```

Comment:
I'm just curious if this method is at all necessary.
What would be the use case for it? Does it need to be present on both API classes? (More of a question to @aalekseev probably...) If we dispose of it, then the `_prepare_document_verification_payload` method would also be redundant, and the docstring could be left nice and neat :)

Reply:
As I understood, that's the main point of changes because for async url is different and returns only response. However, normal verify_document function uses also different url and has validation. Therefore we thought to split payload to another function and use it for both functions.

Reply:
result = apiV2.verify_document_async(params here)
result = apiV2.verify_document(params here) 
result = apiV1.verify_document(params here) 
result = apiV1.verify_document_async(params here) 
Now, API supports all this versions. when we test it. apiV2 is instance of WebApiV2 class, apiV1 is instance of WebApiV1 class

Reply:
the API v1 and v2 are both support async document verification and sync one. not sure how to make it nicely :thinking:

Reply:
Well, it's fine that they support it, the question is - do we *need* it? what's the point? If we aim to build a holistic library that supports every endpoint possible, that's ok - but then it kinda lags behind the goal by far =) If we don't use it, then it feels supporting unused stuff is a waste of time.

Reply:
Ok, from this side of view - we will be using only async endpoint

Reply:
@ykshatroff so are you suggesting to have only one async method, or it is fine to release new version?